### DISCUSSION TITLE
headers key name that contains underscore is missing

### DISCUSSION THREAD

**Discussion by DannyPark1** [vercel/next.js#50899] [App Router] [answered]
setting header with a key name that contains underscore is missing.

```
import ky from "ky";
import { cookies } from "next/headers";

async function Page() {
  const refreshToken = cookies().get("refresh_token").value;
  const accessToken = cookies().get("access_token").value;

  const response = await ky(
    `[API url]`,
    {
      retry: 10,
      hooks: {
        beforeRequest: [
          (request) => {
            request.headers.set(
                "access-token",
              cookies().get("access_token").value
            );
            request.headers.set(
              "refresh_token",
              cookies().get("refresh_token").value
            );
            request.headers.set("x-test-code", "cksdn1");
            request.headers.set("x_test_code", "cksdn2");
          },
        ],
      },
    }
  ).json();
  console.log("response", response);
  return <>test</>;
}

export default Page;
```
<img width="724" alt="스크린샷 2023-06-07 오후 8 59 49" src="https://github.com/vercel/next.js/assets/109129436/4a10e86b-03be-4e81-a2df-50deec1e1433">

as you can see above the image, a key which it contains underscore in key name is all missing. Is this intended?

**Comment by icyJoseph** [comment, marked as answer]
Just learned that NGINX servers drop headers with underscore, https://www.nginx.com/resources/wiki/start/topics/tutorials/config_pitfalls/?highlight=disappearing%20http%20headers#missing-disappearing-http-headers 

Where did you take the screenshot from?
